### ISSUE TITLE
Feature Gate: Deprecate executable meta flag update in bpf loader

### ISSUE THREAD

**Issue by HaoranYi** [solana-labs/solana#34425]
### SIMD

https://github.com/solana-labs/solana/issues/33970

### Description

Using account's owner and loader metadata stored in account's data can tell us whether the account is `executable`. Therefore, we don't need to set account's executable meta when deploy the program by bpf loader. 
When this feature is activated, bpf loader will no longer write *true* to account's executable meta. 

### Feature ID

k6uR1J9VtKJnTukBV2Eo15BEy434MBg8bT6hHQgmU8v

### Activation Method

Single Core Contributor

### Deployment Considerations

NA

**Comment by CriesofCarrots** [comment]
SIMD link does not link to a SIMD

**Comment by HaoranYi** [comment]
> SIMD link does not link to a SIMD

Added

**Comment by billythedummy** [comment]
Did you guys check if there are any currently mainnet-deployed programs that for example check the executable flag on a program AccountInfo before CPI-ing it? They will stop working with any new programs deployed after this feature. Probably also want to check with anchor if they're using the executable flag for any account checks and the like. cc @acheroncrypto

**Comment by acheroncrypto** [comment]
> Probably also want to check with anchor if they're using the executable flag for any account checks and the like.
> 
> cc @acheroncrypto

Anchor uses the `executable` flag in some places but I think that's fine because https://github.com/solana-labs/solana/pull/34194 doesn't seem to affect programs.

**Comment by HaoranYi** [comment]
> Did you guys check if there are any currently mainnet-deployed programs that for example check the executable flag on a program AccountInfo before CPI-ing it? They will stop working with any new programs deployed after this feature. Probably also want to check with anchor if they're using the executable flag for any account checks and the like.
> 
> cc @acheroncrypto
> 
> Update: nvm correct me if i'm wrong but this doesn't change the `AccountInfo` struct that's available to programs since the `executable` field is serialized with the return of the updated function:
> 
> https://github.com/solana-labs/solana/blob/8fde8d26c718edb028a1c77de6471a6e52a7bf03/programs/bpf_loader/src/serialization.rs#L481
> 
> . No plans to deprecate `AccountInfo.executable`, right?

Yes, that's correct. This feature didn't change `AccountInfo.executable`. It only affects the metadata stored in accounts-db.
